ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/provider/DiffBasedWorldStateProvider.java

```diff
@@ -120,40 +119,122 @@ public Optional<WorldState> get(final Hash rootHash, final Hash blockHash) {
   @Override
   public boolean isWorldStateAvailable(final Hash rootHash, final Hash blockHash) {
     return cachedWorldStorageManager.contains(blockHash)
-        || persistedState.blockHash().equals(blockHash)
+        || headWorldState.blockHash().equals(blockHash)
         || worldStateKeyValueStorage.isWorldStateAvailable(rootHash, blockHash);
   }
 
+  /**
+   * Gets the mutable world state based on the provided query parameters.
```

Comment:
nitpick, should be `Gets a mutable world state`.  "The" to me would imply the head worldstate

Reply:
yes clearly better like that